helper_dart/lib/src/insights.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:algolia/algolia.dart';
+import 'package:collection/collection.dart';
+
+import 'algolia_event_service_adapter.dart';
+import 'event_service.dart';
+import 'event_tracker.dart';
+
+/// Insights is the component responsible for sending events related to
+/// the user to personalize his search
+class Insights implements EventTracker {
```

Comment:
Suggestion: maybe have all insights core components in a separate package :D

Reply:
Sure, but I will do it later, this PR is a proof of concept :)